lang/src/lib.rs

```diff
@@ -270,6 +270,20 @@ pub trait InstructionData: Discriminator + AnchorSerialize {
         d.append(&mut self.try_to_vec().expect("Should always serialize"));
         d
     }
+    /// Clears `vec` and writes instruction data to it
+    ///
+    /// We use a Vec<u8> here because of the additional flexibility of re-allocation (only if necessary),
+    /// and because the data field in `Instruction` expects a `Vec<u8>`.
+    fn write_to(&self, mut vec: &mut Vec<u8>) {
+        // Clear vector
+        vec.clear();
+
+        // Write discriminator and then Self
+        Self::DISCRIMINATOR
+            .serialize(&mut vec)
+            .expect("Discriminator is infallibly serializable");
+        self.serialize(&mut vec)
+            .expect("InstructionData should be infallibly serializable");
```

Comment:
Suggestion unintentionally removed the closing brace 😅
```suggestion
            .expect("InstructionData should be infallibly serializable");
        }
```